### PR TITLE
More maven bugfixes

### PR DESCRIPTION
**Pull request by kbarrette**

### CODE REVIEW

File: app/models/package_manager/maven.rb [hunk 1]
```diff
@@ -59,13 +59,19 @@ def self.recent_names
...
         name: name,
-        path: sections.join('/'),
+        path: path,
         groupId: sections[0],
         artifactId: sections[1],
-        versions: versions(h),
+        versions: versions,
       }
+    rescue
+      {}
```
Comment (mikeyoung85) [inline app/models/package_manager/maven.rb]: I think passing back an empty Hash will cause an error further downstream. I don't think the rescue will help.

File: app/models/package_manager/maven.rb [hunk 2]
```diff
@@ -59,13 +59,19 @@ def self.recent_names
 
     def self.project(name)
       sections = name.split(':')
+      path = sections.join('/')
+      versions = versions({ path: path })
+      return {} unless versions.present?
```
Comment (mikeyoung85) [inline app/models/package_manager/maven.rb]: Are we nearly back to the original version of this method?

File: app/models/package_manager/maven.rb [hunk 3]
```diff
@@ -106,7 +112,7 @@ def self.mapping_from_pom_xml(version_xml, depth = 0)
           xml.locate('scm/url/?[0]').first,
           xml.locate('url/?[0]').first
         ),
-        licenses: licenses(xml).join(","),
+        licenses: licenses(version_xml).join(","),
```
Comment (mikeyoung85) [inline app/models/package_manager/maven.rb]: Isn't `xml` the "canonical" xml in this method?

File: app/models/package_manager/maven.rb (hunk 1 above)
Comment (kbarrette) [reply app/models/package_manager/maven.rb]: `Base` checks `project.present?` - this is the same pattern used in other package manager classes to signal "do nothing"

File: app/models/package_manager/maven.rb (hunk 3 above)
Comment (kbarrette) [reply app/models/package_manager/maven.rb]: Nope - we call `version_xml.project` which doesn't include the top-level comment (line 84)

File: app/models/package_manager/maven.rb (hunk 2 above)
Comment (kbarrette) [reply app/models/package_manager/maven.rb]: 🤷‍♂

File: app/models/package_manager/maven.rb (hunk 1 above)
Comment (mikeyoung85) [reply app/models/package_manager/maven.rb]: Could we pass back the Hash with an empty versions if the versions call fails?

**Pull request merged**